build_runner/test/server/serve_handler_test.dart

```diff
@@ -162,6 +164,97 @@ void main() {
+    });
+
+    test('emmits a message to all listners', () async {
+      Function exposedOnConnect;
+      var mockHandlerFactory = (Function onConnect, {protocols}) {
+        exposedOnConnect = onConnect;
+      };
+      var buildUpdatesWebSocketHandler = BuildUpdatesWebSocketHandler(
+          mockHandlerFactory);
+
+      var streamControllerFirst1 = StreamController<List<int>>();
+      var streamControllerFirst2 = StreamController<List<int>>();
+      var serverChannelFirst = WebSocketChannel(StreamChannel(streamControllerFirst1.stream, streamControllerFirst2.sink), serverSide: true);
+      var clientChannelFirst = WebSocketChannel(StreamChannel(streamControllerFirst2.stream, streamControllerFirst1.sink), serverSide: false);
+
+      var streamControllerSecond1 = StreamController<List<int>>();
+      var streamControllerSecond2 = StreamController<List<int>>();
+      var serverChannelSecond = WebSocketChannel(StreamChannel(streamControllerSecond1.stream, streamControllerSecond2.sink), serverSide: true);
+      var clientChannelSecond = WebSocketChannel(StreamChannel(streamControllerSecond2.stream, streamControllerSecond1.sink), serverSide: false);
+
+      var deferredExpect1 = clientChannelFirst.stream.single.then((value) => expect(value, 'update'));
+      var deferredExpect2 = clientChannelSecond.stream.single.then((value) => expect(value, 'update'));
+      exposedOnConnect(serverChannelFirst, '');
+      exposedOnConnect(serverChannelSecond, '');
+      buildUpdatesWebSocketHandler.emitUpdateMessage(null);
+      await clientChannelFirst.sink.close();
+      await clientChannelSecond.sink.close();
+      await deferredExpect1;
+      await deferredExpect2;
+    });
+
+    test('deletes listners on disconect', () async {
+      Function exposedOnConnect;
+      var mockHandlerFactory = (Function onConnect, {protocols}) {
+        exposedOnConnect = onConnect;
+      };
+      var buildUpdatesWebSocketHandler = BuildUpdatesWebSocketHandler(
+          mockHandlerFactory);
+
+      var streamControllerFirst1 = StreamController<List<int>>();
+      var streamControllerFirst2 = StreamController<List<int>>();
+      var serverChannelFirst = WebSocketChannel(StreamChannel(streamControllerFirst1.stream, streamControllerFirst2.sink), serverSide: true);
+      var clientChannelFirst = WebSocketChannel(StreamChannel(streamControllerFirst2.stream, streamControllerFirst1.sink), serverSide: false);
+
+      var streamControllerSecond1 = StreamController<List<int>>();
+      var streamControllerSecond2 = StreamController<List<int>>();
+      var serverChannelSecond = WebSocketChannel(StreamChannel(streamControllerSecond1.stream, streamControllerSecond2.sink), serverSide: true);
+      var clientChannelSecond = WebSocketChannel(StreamChannel(streamControllerSecond2.stream, streamControllerSecond1.sink), serverSide: false);
+
+      var deferredExpect1 = clientChannelFirst.stream.toList().then((value) => expect(value, ['update', 'update']));
```

Comment:
Instead of this you can use the [built in stream matchers](https://github.com/dart-lang/test#stream-matchers)